src/index.js

```diff
@@ -404,6 +413,35 @@ export type EdgePluginData = {
   setItem(pluginId: string, itemId: string, value: string): Promise<mixed>
 }
 
+// Currencies supported by various exchange-rate providers:
+export type EdgeExchangeCurrencies = {
+  [currencyCode: string]: {
+    exchanges: Array<string>
+  }
+}
+
+export type EdgeExchangeQuoteOptions = {
+  fromCurrencyCode: string,
+  fromWallet: EdgeCurrencyWallet,
+  nativeAmount: string,
+  quoteFor: 'from' | 'to',
+  toCurrencyCode: string,
+  toWallet: EdgeCurrencyWallet
+}
+
+export type EdgeExchangeQuote = {
+  +expirationDate: Date,
+  +fromNativeAmount: string,
+  +toNativeAmount: string,
+
+  +exchangeService: string,
+  +quoteId: string,
+  +quoteUri: string,
```

Comment:
TODO: Put this into `orderUri` in the metadata on approval, and remove it from the quote.